src/librustc_mir/borrow_check/nll/region_infer/error_reporting/outlives_suggestion.rs

```diff
@@ -0,0 +1,254 @@
+            | RegionNameSource::Static => {
+                debug!("Region {:?} is suggestable", name);
+                true
+            }
+
+            // Don't give suggestions for upvars, closure return types, or other unnamable
+            // regions.
+            RegionNameSource::SynthesizedFreeEnvRegion(..)
+            | RegionNameSource::CannotMatchHirTy(..)
+            | RegionNameSource::MatchedHirTy(..)
+            | RegionNameSource::MatchedAdtAndSegment(..)
+            | RegionNameSource::AnonRegionFromUpvar(..)
+            | RegionNameSource::AnonRegionFromOutput(..)
+            | RegionNameSource::AnonRegionFromYieldTy(..) => {
+                debug!("Region {:?} is NOT suggestable", name);
+                false
+            }
+        }
+    }
+
+    /// Returns a name for the region if it is suggestable. See `region_name_is_suggestable`.
+    fn region_vid_to_name(
+        &self,
+        errctx: &ErrorReportingCtx<'_, '_, '_>,
+        renctx: &mut RegionErrorNamingCtx,
+        region: RegionVid,
+    ) -> Option<RegionName> {
+        errctx
+            .rinfcx
+            .give_region_a_name(errctx, renctx, region)
+            .filter(Self::region_name_is_suggestable)
+    }
+
+    /// Add the outlives constraint `fr: outlived_fr` to the set of constraints we need to suggest.
+    crate fn collect_constraint(&mut self, fr: RegionVid, outlived_fr: RegionVid) {
+        debug!("Collected {:?}: {:?}", fr, outlived_fr);
+
+        // Add to set of constraints for final help note.
+        if let Some(ref mut outlived_frs) = self.constraints_to_add.get_mut(&fr) {
+            outlived_frs.push(outlived_fr);
+        } else {
+            self.constraints_to_add.insert(fr, vec![outlived_fr]);
+        }
+    }
+
+    /// Emit an intermediate note on the given `Diagnostic` if the involved regions are
+    /// suggestable.
+    crate fn intermediate_suggestion(
+        &mut self,
+        errctx: &ErrorReportingCtx<'_, '_, '_>,
+        errci: &ErrorConstraintInfo,
+        renctx: &mut RegionErrorNamingCtx,
+        diag: &mut DiagnosticBuilder<'_>,
+    ) {
+        // Emit an intermediate note.
+        let fr_name = self.region_vid_to_name(errctx, renctx, errci.fr);
+        let outlived_fr_name = self.region_vid_to_name(errctx, renctx, errci.outlived_fr);
+
+        match (fr_name, outlived_fr_name) {
+            (Some(fr_name), Some(outlived_fr_name)) => {
+                if let RegionNameSource::Static = outlived_fr_name.source {
+                    diag.help(&format!(
+                        "Consider replacing `{}` with `'static`", fr_name
+                    ));
+                } else {
+                    diag.help(&format!(
+                        "Consider adding the following bound: `{}: {}`",
+                        fr_name, outlived_fr_name
+                    ));
+                }
+            }
+
+            // Not both suggestable
+            _ => {}
```

Comment:
You could use `if let (Some(fr_name), Some(outlived_fr_name)) = (fr_name, outlived_fr_name) {`